cpp/src/neighbors/specializations/detail/ball_cover_lowdim_pass_two_3d.cu

```diff
@@ -15,11 +15,9 @@
  */
 
 #include <cstdint>
+#include <raft/neighbors/specializations.cuh>
 #include <raft/spatial/knn/detail/ball_cover/registers.cuh>
 
-// TODO: Change this to proper specializations after FAISS is removed
```

Comment:
Yeah, so @divyegala had a similar question and I can understand why it looks confusing here. Imagine the inital goals of RAFT's building individual (and individually optional) packages for `nn` and `distance`. We had users like cugraph who didn't want a FAISS dependency but users like cuml who didn't care. We also needed to be careful to keep the faiss dependency out of `pylibraft` for the same reasons. As a result, we made it such that `libraft-nn` had no dependency on `libraft-distance` and vice versa. A user *could* configure their build to compile and install both, in which case they could share specializations which were already compiled. 

However, if a user only built `libraft-nn` or `libraft-distance` individually, we needed to be careful that any APIs they were using that had specializations headers would not ignore compilingn specializations for the functions which they had not pre-compiled already. This meant we had to be very careful about keeping them separate. As a "standard practice" we want users to only have to worry about including `raft/<namespace>/specializations.cuh` for the namespaces they are using to ease their use of raft, but if we always included all of the namespaces in all of the spcecializations files, they'd get linking errors if they didn't always compile both libraries. 

So what we did was deprecate the `spatial/` directory and all of its contents. Since we now have our own brute-force knn not based on faiss, we moved those specializtions over to the libraft-distance package. The only two APIs that remained in `libraft-nn` were random ball cover and the fused l2 1-nn (which specifically dependened on the faiss k-selection device functions). We isolated those headers down in `spatial/knn` to separate them and didn't include them in `neighrbos/specializations.cuh`. That way special care could be taken only for those two APIs instead of having to split out the majority. Now that we're consolidating these apis and have removed the need to compile or link against the faiss library (though kept some choice headers in the meantime which are copied into the codebase), we can merge the two together, which also means merging the specializations. Underneath, @benfred has updated the `spatial/knn` headers to just use the `neighbors/` headers since those are being used now anyways. Next steps (which will be done in 23.06) will be to move all the needed `spatial/knn` headers over to `neighbors/` and finally remove those which are no longer needed.

Reply:
Got it, thanks for that context. It helps a lot to understand the various moving pieces at play (including those in PRs that I haven't actively reviewed) to know why some of these changes are all happening in concert.